randomUsernameAndPassword/assignment1/Q2.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+class Node:
+  def __init__(self, values):
+    assert(values)
+    head, tail = values[0], values[1:]
+    self.data = head
+    if tail:
+      self.next = Node(tail)
+    else:
+      self.next = None
+
+  def get_next(self):
+    return self.next
+
+  def get_data(self):
+    return self.data
+
+def get_kth_last(head, k):
+  if (head == None or k < 0):
+    return None
+  
+  elem = head
+  counter = 0
+  
+  while (elem is not None):
```

Comment:
Do you think you can add a __len__ method to the Node?

Reply:
Fixed